src/ode_solver/sundials.rs

```diff
@@ -407,30 +408,30 @@ where
             IDA_SUCCESS => Ok(OdeSolverStopReason::InternalTimestep),
             IDA_TSTOP_RETURN => Ok(OdeSolverStopReason::TstopReached),
             IDA_ROOT_RETURN => Ok(OdeSolverStopReason::RootFound(state.t)),
-            IDA_MEM_NULL => Err(anyhow!("The ida_mem argument was NULL.")),
-            IDA_ILL_INPUT => Err(anyhow!("One of the inputs to IDASolve() was illegal, or some other input to the solver was either illegal or missing.")),
-            IDA_TOO_MUCH_WORK => Err(anyhow!("The solver took mxstep internal steps but could not reach tout.")),
-            IDA_TOO_MUCH_ACC => Err(anyhow!("The solver could not satisfy the accuracy demanded by the user for some internal step.")),
-            IDA_ERR_FAIL => Err(anyhow!("Error test failures occurred too many times (MXNEF = 10) during one internal time step or occurred with.")),
-            IDA_CONV_FAIL => Err(anyhow!("Convergence test failures occurred too many times (MXNCF = 10) during one internal time step or occurred with.")),
-            IDA_LINIT_FAIL => Err(anyhow!("The linear solver’s initialization function failed.")),
-            IDA_LSETUP_FAIL => Err(anyhow!("The linear solver’s setup function failed in an unrecoverable manner.")),
-            IDA_LSOLVE_FAIL => Err(anyhow!("The linear solver’s solve function failed in an unrecoverable manner.")),
-            IDA_CONSTR_FAIL => Err(anyhow!("The inequality constraints were violated and the solver was unable to recover.")),
-            IDA_REP_RES_ERR => Err(anyhow!("The user’s residual function repeatedly returned a recoverable error flag, but the solver was unable to recover.")),
-            IDA_RES_FAIL => Err(anyhow!("The user’s residual function returned a nonrecoverable error flag.")),
-            IDA_RTFUNC_FAIL => Err(anyhow!("The rootfinding function failed.")),
-            _ => Err(anyhow!("Unknown error")),
+            IDA_MEM_NULL => Err(other_error!("The ida_mem argument was NULL.")),
```

Comment:
should these all be `OdeSolverError::SundialsError` as well?

Reply:
Sure, I just didn't know if the those from `IDA` was regarded as such.

Reply:
Yup, IDA is a particular solver within the Sundials solver suite

Reply:
Fixed